src/dex-ui/pages/dao/DAOProposals/Forms/DAOTokenTransferDetailsForm.tsx

```diff
@@ -35,34 +38,37 @@ export function DAOTokenTransferDetailsForm() {
   watch("tokenId");
   const accountTokenBalancesQueryResults = useAccountTokenBalances(safeAccountId ?? "");
   const { data: tokenBalances } = accountTokenBalancesQueryResults;
-  const selectedAsset = getValues().tokenId
-    ? tokenBalances?.find(
-        (asset: TokenBalance) =>
-          asset.tokenId === getValues().tokenId || (getValues().tokenId === HBARTokenId && asset.symbol === HBARSymbol)
-      )
-    : undefined;
+  const { tokenId } = getValues();
+  const selectedAsset =
+    isEmpty(tokenId) || isNil(tokenId)
+      ? undefined
+      : tokenBalances?.find(
+          (asset: TokenBalance) => asset.tokenId === tokenId || (tokenId === HBARTokenId && asset.symbol === HBARSymbol)
+        );
 
   const { handleMaxButtonClicked, handleHalfButtonClicked } = useHalfMaxButtons(
-    selectedAsset?.balance ?? 0,
-    (amount: number) => setValue("amount", amount)
+    String(selectedAsset?.balance ?? 0),
+    (amount: string | undefined) => setValue("amount", amount)
   );
 
+  const { handleTokenInputChangeWithPattern } = useFormTokenInputPattern((value: string) => setValue("amount", value));
+
   if (proposalType !== DAOProposalType.TokenTransfer) {
     setValue("type", DAOProposalType.TokenTransfer);
   }
 
-  function validateAmount(value: number) {
-    const tokenId = getValues().tokenId;
-    const selectedAsset = tokenBalances?.find(
-      (asset: TokenBalance) => asset.tokenId === tokenId || (tokenId === HBARTokenId && asset.symbol === HBARSymbol)
-    );
+  function validateAmount(value: string | undefined) {
```

Comment:
If we pass very small number or a number which is not rounding off to 8 decimals. we are seeing error on the next screen, Can we add a condition in validation/regex to prevent this scenario?
<img width="722" alt="Screenshot 2023-07-28 at 11 58 04 PM" src="https://github.com/hashgraph/hedera-accelerator-defi-dex-ui/assets/132666177/8c037ec1-08e4-451f-9e43-01cb777cf19d">

Reply:
I sent a message in Teams to confirm the UX to prevent this scenario across the app. I will address this in a subsequent PR.